### DISCUSSION TITLE
Bra verktyg

### DISCUSSION THREAD

**Discussion by salgo60** [salgo60/Svenskabadplatser#46] [Show and tell]
Kopplar man Wikidata till OSM kan man lägga in följande rad i sin WIkidata profil i [filen common,.js](https://www.wikidata.org/wiki/User:Salgo60/common.js)

`mw.loader.load( '//www.wikidata.org/w/index.php?title=User:Mxn/overpass.js&action=raw&ctype=text/javascript' ); `

==> att en iFrame visas med OSM objektet (görs en Overpass sökning)

* Exempel [Q106708615](https://www.wikidata.org/wiki/Q106708615) som har koppling [OSM way/943540539](https://www.openstreetmap.org/way/943540539)

sökning sker med [Overpass](http://overpass-turbo.eu/s/17p7)

```
[out:json][timeout:25];
nwr["wikidata"="Q106708615"];
out body geom qt;
```

![image](https://user-images.githubusercontent.com/14206509/118400629-b71b4500-b662-11eb-95c1-86dd0edccba1.png)

**Comment by salgo60** [comment]
* [OpenSwitchMaps](https://github.com/tankaru/OpenSwitchMaps)

trevlig browser plugin som hoppar mellan olika kartor snabbt. Bra om man skall kartera en strand och vill kolla på flera
![image](https://user-images.githubusercontent.com/14206509/118400806-838cea80-b663-11eb-9ca4-614d13657b56.png)

**Comment by soleildeminuit** [comment]
Tack, väldigt användbart att få tillgång till OSM-vyn

**Reply by salgo60** [reply]
@anderselias och direkt börjar man fundera när man kollar på [Q106653102](https://alpha.osm.wikidata.link/map/18/59.43824/17.99569?item=Q106653102)
![image](https://user-images.githubusercontent.com/14206509/118566909-12d5f300-b775-11eb-9f88-6e3508122dcc.png)


1. Vad vill vi ha från Havs- och vattenmyndighetens i
    1. koordinat badet
    1. yta badet 
    2. koordinat mätpunkt
    3. beach?
    4. mått på bakterier, algblommning ?!?!?....
    5. väderidentifierare till SMHIs API?!?!?
    5. andra miljöpåverkande saker i omgivningen
    6. vilket vatten ett bad finns vid identifierad med ett sjö id och bev WDid [SPARQL](https://w.wiki/3LnN)
       1. yta geojson vattnet  
    7. kommunen identifierad med kommunkod och/eller Wikidata Q nummer [SPARQL](https://w.wiki/3LnL) se https://github.com/salgo60/Svenskabadplatser/issues/51
    8. varför "delar man upp" ett bad ex. [SE0110180000001841](https://badplatsen.havochvatten.se/badplatsen/karta/#/bath/SE0110180000001841) och [SE0110180000001840](https://badplatsen.havochvatten.se/badplatsen/karta/#/bath/SE0110180000001840)  se [osm.wikidata.link Q106708722](https://alpha.osm.wikidata.link/map/18/59.35679/17.84087?item=Q106708722) vi har WD [Q10576216](https://www.wikidata.org/wiki/Q10576216) och [Q106708722](https://www.wikidata.org/wiki/Q106708722) och i OSM har vi [relation/12465474](https://www.openstreetmap.org/relation/12465474)
1. vad är ett bad i Wikidata
2. vad är ett bad i OSM
    1. en yta [Tag:leisure=swimming_area](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dswimming_area) ev. lika med Havs- och vattenmyndighetens yra
    1. en [Tag:natural=beach](https://wiki.openstreetmap.org/wiki/Tag:natural%3Dbeach)
    1. ???  
 1. Vad vill vi ha från kommunerna
    1. publik identifierare som säger att badet är samma som  Havs- och vattenmyndighetens
    2. url hemsida
    2. hur man felrapporterar saker på ett bad
    3. tillgänglighetsinformation
    4. hur man åker kommunalt dit
        1.  gärna hållplatsid i något API för kommunala resor...
    2. ?!?!?

**Comment by salgo60** [comment]
Tool to compare WIkidata and OSM coordinates

https://alpha.osm.wikidata.link/map/16/59.43910/18.00075

below with public beaches selected is created by [edwardbetts](https://twitter.com/edwardbetts) [edwardbetts.com](http://edwardbetts.com/) [GITHUB](https://github.com/EdwardBetts)

![image](https://user-images.githubusercontent.com/14206509/118558748-5cb7dc80-b767-11eb-8a77-69bce54562fb.png)


cc: @anderselias @dpriskorn

**Comment by soleildeminuit** [comment]
Tool for extracting OSM polygons
http://polygons.openstreetmap.fr/

![image](https://user-images.githubusercontent.com/10654578/118560330-9b4e9680-b769-11eb-924a-ff2f5cbdb804.png)

**Comment by salgo60** [comment]
**[Wikidata to OpenStreetMap](https://osm.wikidata.link/tagged/)** skapat av samma kille som innan [edwardbetts](https://twitter.com/edwardbetts) [edwardbetts.com](http://edwardbetts.com/)

Hämtar ut om ett OSM objekt finns med koppling till ett Wikidata objekt 
![image](https://user-images.githubusercontent.com/14206509/118582757-2e9cc180-b794-11eb-9398-8d75fc903e5d.png)

* [Notebook](https://github.com/salgo60/Svenskabadplatser/blob/main/Jupyter/OSM_Wikidata_Bathwater.ipynb) som kör igenom alla bad i projektet ==> 2327 av 2755 85% saknar Wikidata tag i OSM
  * exempel med [Q106710481](https://osm.wikidata.link/tagged/api/item/Q106710481) och utan [Q106708833](https://osm.wikidata.link/tagged/api/item/Q106708833) 
* GITHUB [EdwardBetts/osm-wikidata-tagged](https://github.com/EdwardBetts/osm-wikidata-tagged)

**Comment by salgo60** [comment]
Fick [denna SPARQL](https://w.wiki/3MA7) från Larske som hittar svenska sjöar nära [Q106653102](https://www.wikidata.org/wiki/Q106709055) 
* hade förväntat mig att hitta [Ösjön i Wikidata](https://www.openstreetmap.org/relation/77705#map=13/60.4937/15.5743) men den finns inte

```
#defaultView:Map
SELECT ?bad ?badLabel ?nearbys ?nearbysLabel ?nearbyscoord ?dist {
  VALUES ?bad { wd:Q106709055 }
  ?bad wdt:P625 ?badcoord .
  ?nearbys wdt:P31 wd:Q23397 .
  ?nearbys wdt:P17 wd:Q34 .
  ?nearbys wdt:P625 ?nearbyscoord .
  BIND(geof:distance(?badcoord,?nearbyscoord) AS ?dist)
  FILTER(?dist<10)
  SERVICE wikibase:label { bd:serviceParam wikibase:language "sv,en". }
}    
```
![image](https://user-images.githubusercontent.com/14206509/118771355-bd880780-b882-11eb-8300-e3b27331b16e.png)

* [OSM Ösjön](https://www.openstreetmap.org/relation/77705#map=13/60.4937/15.5743)
   * [osm.wikidata.link/map/11/60.53950/15.71564](https://alpha.osm.wikidata.link/map/11/60.53950/15.71564)
![image](https://user-images.githubusercontent.com/14206509/118771675-1ce61780-b883-11eb-81a4-040e32d9cf3f.png)

![image](https://user-images.githubusercontent.com/14206509/118771948-71899280-b883-11eb-9f92-2a52ac97d3ea.png)

**Comment by salgo60** [comment]
Browser plug.in [Entity Explosion](https://www.wikidata.org/wiki/Wikidata:Entity_Explosion) som använder Wikidata för att navigera mellan [Havs badplatser](https://badplatsen.havochvatten.se/badplatsen/karta/#/bath/SE0110120000001680), [Badkartan](https://www.badkartan.se/049327/%C3%96stersj%C3%B6viken), [Kommunernas webplats](https://www.varmdo.se/upplevaochgora/naturochfriluftsliv/badplatserivarmdo/ostersjoviken.4.18c983316e0536cb189a288.html), [Instagram locations](https://www.instagram.com/explore/locations/252758059/) 

* [video](https://youtu.be/OdxMCHdF23k)

* nya egenskaper i Wikidata 
  * [EIONET bathwater identification](https://www.wikidata.org/wiki/Property:P9616) Property:P9616
  * [Badkartan](https://www.wikidata.org/wiki/Property:P9615) Property:P9615
  
![image](https://user-images.githubusercontent.com/14206509/120382289-c1586700-c323-11eb-981c-f8b950d5185a.png)

![image](https://user-images.githubusercontent.com/14206509/120382552-11cfc480-c324-11eb-938e-d0da25621cd3.png)

![image](https://user-images.githubusercontent.com/14206509/120382754-53f90600-c324-11eb-9efe-204fa39d42c2.png)

**Comment by salgo60** [comment]
@soleildeminuit Wikidata and R
* [Wikidata and R: a perfect pair](https://diff.wikimedia.org/2021/06/03/wikidata-and-r-a-perfect-pair/)
* [github.com/TS404/WikidataR](https://github.com/TS404/WikidataR)

**Comment by salgo60** [comment]
OSM har nu en SPARQL som fungerar där man kan göra federation med Wikidata [se exempel ](https://sophox.org/#%23%20leisure%20swimming_area%20in%20OSM%20with%20Wikidata%20tag%0A%23defaultView%3AMap%0ASELECT%20DISTINCT%20%3Fosm%20%3Fcoordinates%20%3FWD%20%3FwdLabel%20%3FHAVid%20%3FHavAPI%20%3FHavAPI2%20%3FHav%20WHERE%20%7B%0A%20%20%23%20Prevent%20optimizer%20from%20querying%20for%20everything%20tagged%20with%20wikidata%3D%2a%20before%20querying%20the%20service.%0A%20%20hint%3AQuery%20hint%3Aoptimizer%20%22None%22%20.%0A%20%20%0A%20%20%23%20Query%20OpenStreetMap%20for%20leisure%20swimming_area%0A%20%20%3Fosm%20osmt%3Aleisure%20%22swimming_area%22.%0A%20%20%3Fosm%20osmt%3Awikidata%20%3Fwd.%0A%20%20SERVICE%20%3Chttps%3A%2F%2Fquery.wikidata.org%2Fsparql%3E%20%7B%0A%20%20%20%20%3Fwd%20wdt%3AP9616%20%3FHAVid.%0A%20%20%20%20%3Fwd%20wdt%3AP17%20wd%3AQ34%0A%20%20%20%20%20%20%20%20%0A%20%20BIND%20%28URI%28CONCAT%28%22https%3A%2F%2Fbadplatsen.havochvatten.se%2Fbadplatsen%2Fapi%2Ftestlocationprofile%2F%22%2C%3FHAVid%29%29%20AS%20%3FHavAPI%29%0A%20%20BIND%20%28URI%28CONCAT%28%22https%3A%2F%2Fbadplatsen.havochvatten.se%2Fbadplatsen%2Fapi%2Fdetail%2F%22%2C%3FHAVid%29%29%20AS%20%3FHavAPI2%29%0A%20%20BIND%20%28URI%28CONCAT%28%22https%3A%2F%2Fbadplatsen.havochvatten.se%2Fbadplatsen%2Fkarta%2F%23%2Fbath%2F%22%2C%3FHAVid%29%29%20AS%20%3FHav%29%0A%20%20%7D%20%20%0A%20%0A%20%20%0A%20%20%23%20Get%20coordinates%20to%20display%20on%20a%20map%0A%20%20%3Fosm%20osmm%3Aloc%20%3Fcoordinates.%0A%7D)
  
![image](https://user-images.githubusercontent.com/14206509/122372065-a6f7cd80-cf60-11eb-9227-ab5e5f783596.png)

```
# leisure swimming_area in OSM with Wikidata tag
#defaultView:Map
SELECT DISTINCT ?osm ?coordinates ?WD ?wdLabel ?HAVid ?HavAPI ?HavAPI2 ?Hav WHERE {
  # Prevent optimizer from querying for everything tagged with wikidata=* before querying the service.
  hint:Query hint:optimizer "None" .
  
  # Query OpenStreetMap for leisure swimming_area
  ?osm osmt:leisure "swimming_area".
  ?osm osmt:wikidata ?wd.
  SERVICE <https://query.wikidata.org/sparql> {
    ?wd wdt:P9616 ?HAVid.
    ?wd wdt:P17 wd:Q34
        
  BIND (URI(CONCAT("https://badplatsen.havochvatten.se/badplatsen/api/testlocationprofile/",?HAVid)) AS ?HavAPI)
  BIND (URI(CONCAT("https://badplatsen.havochvatten.se/badplatsen/api/detail/",?HAVid)) AS ?HavAPI2)
  BIND (URI(CONCAT("https://badplatsen.havochvatten.se/badplatsen/karta/#/bath/",?HAVid)) AS ?Hav)
  }  
  
  # Get coordinates to display on a map
  ?osm osmm:loc ?coordinates.
} 
```
